airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/request_option.py

```diff
@@ -25,21 +24,10 @@ class RequestOption:
     Describes an option to set on a request
 
     Attributes:
+        field_name (str): Describes the name of the parameter to inject. None if option_type == path. Required otherwise.
         inject_into (RequestOptionType): Describes where in the HTTP request to inject the parameter
-        field_name (Optional[str]): Describes the name of the parameter to inject. None if option_type == path. Required otherwise.
     """
 
+    field_name: str
     inject_into: RequestOptionType
     parameters: InitVar[Mapping[str, Any]]
-    field_name: Optional[str] = None
-
-    def __post_init__(self, parameters: Mapping[str, Any]):
```

Comment:
In the original implementation, only path RequestOption could skip `field_name`. Now that its been extracted, field_name is required and we can get rid of this entire validation